src/uucore/src/lib/features/checksum.rs

```diff
@@ -335,35 +299,184 @@ fn determine_regex(
     input_is_stdin: bool,
     lines: &[String],
 ) -> UResult<(Regex, bool)> {
-    let algo_based_regex = Regex::new(ALGO_BASED_REGEX).unwrap();
-    let double_space_regex = Regex::new(DOUBLE_SPACE_REGEX).unwrap();
-    let single_space_regex = Regex::new(SINGLE_SPACE_REGEX).unwrap();
+    let regexes = [
+        (Regex::new(ALGO_BASED_REGEX).unwrap(), true),
+        (Regex::new(DOUBLE_SPACE_REGEX).unwrap(), false),
+        (Regex::new(SINGLE_SPACE_REGEX).unwrap(), false),
+        (Regex::new(ALGO_BASED_REGEX_BASE64).unwrap(), true),
+    ];
 
     for line in lines {
         let line_trim = line.trim();
-        if algo_based_regex.is_match(line_trim) {
-            return Ok((algo_based_regex, true));
-        } else if double_space_regex.is_match(line_trim) {
-            return Ok((double_space_regex, false));
-        } else if single_space_regex.is_match(line_trim) {
-            return Ok((single_space_regex, false));
+        for (regex, is_algo_based) in &regexes {
+            if regex.is_match(line_trim) {
+                return Ok((regex.clone(), *is_algo_based));
+            }
+        }
+    }
+    Err(ChecksumError::NoProperlyFormattedChecksumLinesFound {
+        filename: get_filename_for_output(filename, input_is_stdin),
+    }
+    .into())
+}
+
+// Converts bytes to a hexadecimal string
+fn bytes_to_hex(bytes: &[u8]) -> String {
+    bytes
+        .iter()
+        .map(|byte| format!("{:02x}", byte))
+        .collect::<Vec<String>>()
+        .join("")
+}
+
+// Gets the expected checksum
+fn get_expected_checksum(
+    filename: &str,
+    caps: &regex::Captures,
+    chosen_regex: &Regex,
+) -> UResult<String> {
+    if chosen_regex.as_str() == ALGO_BASED_REGEX_BASE64 {
+        let ck = caps.name("checksum").unwrap().as_str();
+        match BASE64.decode(ck.as_bytes()) {
+            Ok(decoded_bytes) => {
+                match std::str::from_utf8(&decoded_bytes) {
+                    Ok(decoded_str) => Ok(decoded_str.to_string()),
+                    Err(_) => Ok(bytes_to_hex(&decoded_bytes)), // Handle as raw bytes if not valid UTF-8
+                }
+            }
+            Err(_) => Err(Box::new(
+                ChecksumError::NoProperlyFormattedChecksumLinesFound {
+                    filename: (&filename).to_string(),
+                },
+            )),
         }
+    } else {
+        Ok(caps.name("checksum").unwrap().as_str().to_string())
     }
+}
 
-    Err(
-        ChecksumError::NoProperlyFormattedChecksumLinesFound(get_filename_for_output(
-            filename,
-            input_is_stdin,
-        ))
-        .into(),
-    )
+/// Returns a reader that reads from the specified file, or from stdin if `filename_to_check` is "-".
+fn get_file_to_check(
+    filename_to_check: &str,
```

Comment:
I think you can shorten `filename_to_check` to `filename` because there is only filename.